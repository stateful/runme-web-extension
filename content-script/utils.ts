export interface LinkDetails {
  fileToOpen: string
  repository: string
}

export function getLinkDetails (pathname = document.location.pathname): LinkDetails {
    let fileToOpen = 'README.md'
    const [orgId, projectName, tree, sha, ...pathToFile] = pathname.slice(1).split('/')

    if (pathToFile.length) {
        fileToOpen = pathToFile.join('/') + ((pathToFile.slice(-1)[0] || '').endsWith('.md') ? '' : '/README.md')
    }

    return {
        repository: `https://github.com/${orgId}/${projectName}.git`,
        fileToOpen
    }
}

export function getVSCodeHref (repository: string, fileToOpen: string) {
  return `https://runme.dev/api/runme?repository=${encodeURIComponent(repository)}&fileToOpen=${fileToOpen}`
}
