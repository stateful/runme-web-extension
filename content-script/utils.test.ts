import { test, expect, vi } from 'vitest'

import { getLinkDetails } from './utils.js'

test('getLinkDetails', () => {
    globalThis.document = <any>{ location: { pathname: '/webdriverio/webdriverio' } }
    expect(getLinkDetails()).toMatchSnapshot()

    globalThis.document = <any>{ location: { pathname: '/webdriverio/webdriverio/tree/main/packages/wdio-jasmine-framework' } }
    expect(getLinkDetails()).toMatchSnapshot()

    globalThis.document = <any>{ location: { pathname: '/webdriverio/webdriverio/blob/main/packages/wdio-jasmine-framework/README.md' } }
    expect(getLinkDetails()).toMatchSnapshot()

    globalThis.document = <any>{ location: { pathname: '/webdriverio/webdriverio/blob/main/CODE_OF_CONDUCT.md' } }
    expect(getLinkDetails()).toMatchSnapshot()
})