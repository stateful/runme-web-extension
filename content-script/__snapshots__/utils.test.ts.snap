// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getLinkDetails 1`] = `
{
  "fileToOpen": "README.md",
  "repository": "https://github.com/webdriverio/webdriverio.git",
}
`;

exports[`getLinkDetails 2`] = `
{
  "fileToOpen": "packages/wdio-jasmine-framework/README.md",
  "repository": "https://github.com/webdriverio/webdriverio.git",
}
`;

exports[`getLinkDetails 3`] = `
{
  "fileToOpen": "packages/wdio-jasmine-framework/README.md",
  "repository": "https://github.com/webdriverio/webdriverio.git",
}
`;

exports[`getLinkDetails 4`] = `
{
  "fileToOpen": "CODE_OF_CONDUCT.md",
  "repository": "https://github.com/webdriverio/webdriverio.git",
}
`;
